### PR TITLE
Rails 6 upgrade

### PR DESCRIPTION
**Pull request by cdmo**
#575 

Following https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-5-2-to-rails-6-0

* App mostly works
* [Lots of problems reported when running `bin/rake`](https://github.com/psu-libraries/psulib_blacklight/files/4654333/rails.6.upgrade.txt)
* Marc display is not working

Been wrestling with why zeitwerk autoloader is seemingly causing problems with blacklight marc things.

### CODE REVIEW

**Comment by cdmo** [comment]
Couple PRs into blacklight and blacklight marc to resolve zeitwerk problems

**Comment by cdmo** [comment]
Using the `:classic` autoloader will clear up the autoload crashes associated with blacklight marc, but then there's this problem too:

https://github.com/projectblacklight/blacklight-marc/pull/87

**Comment by cdmo** [comment]
blacklight-marc should be good to go now, will try again

**Comment by cdmo** [comment]
the blacklight-marc work cleared the way for us to get to Rails 6 🎉

File: config/application.rb [hunk 1]
```diff
@@ -11,7 +11,10 @@
 module Psulcat
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 5.2
+    config.load_defaults '6.0'
+
+    # Should be zeitwerk, but we need to upgrade to Blacklight > 7.9 first. Once we have that we can delete this line.
```
Comment (banukutlu) [inline config/application.rb:16]: thanks for the note!

**Pull request merged**